types/staking.go

```diff
@@ -40,9 +40,9 @@ type BondStatus int32
 
 // staking constants
 const (
-	Unbonded  BondStatus = 1
-	Unbonding BondStatus = 2
-	Bonded    BondStatus = 3
+	Unbonded  BondStatus = 0
+	Unbonding BondStatus = 1
+	Bonded    BondStatus = 2
```

Comment:
Fixing some stuff lost in translation during the proto migration.

https://github.com/cosmos/cosmos-sdk/commit/53bf2271d5bac054a8f74723732f21055c1b72d4#diff-f54554903608b8b89649f532c8f1a78cR43-R45

Reply:
(unless we prefer not having 0 as BondStatus, in which case I can also make a migration script for that)

Reply:
Shouldn't these be protobuf enums? In which case we want 0 to be UNSPECIFIED

Reply:
yeah, i had exactly the same idea, started here: https://github.com/cosmos/cosmos-sdk/compare/am-migrate-followup...am-migrate-followup-staking?expand=1.

The only blocker is an import cycle: `x/staking/exported` depends on `x/staking/types` for BondStatus, but `x/staking/types` also imports `x/staking/exported` for interface impl assertion.

A solution would be to put `BondStatus` in `cosmos.base`, but that also seems weird.

Reply:
Switching BondStatus is imo not in the scope of this PR, so I created another one: #7499 .

I also reverted BondStatus to use 1,2,3 instead of 0,1,2.